File: Dummy/hawkshaw/library/features/media/files/FileExplorerMediaStoreKt.java
package com.hawkshaw.library.features.media.files;

import X2.q;
import Y1.K;
import android.content.ContentResolver;
import android.content.ContentUris;
import android.database.Cursor;
import android.net.Uri;
import android.os.Build;
import android.provider.MediaStore;
import android.util.Log;
import java.util.ArrayList;
import java.util.List;

public final class FileExplorerMediaStoreKt {
    private static final String TAG = "FileExplorer";

    public static final List<MediaStoreFile> getMediaStoreFiles(ContentResolver contentResolver) {
        int i5;
        String str;
        ArrayList arrayList;
        K.n(contentResolver, "contentResolver");
        int i6 = Build.VERSION.SDK_INT;
        q qVar = q.f2602D;
        if (i6 >= 29) {
            Uri contentUri = MediaStore.Files.getContentUri("external");
            ArrayList arrayList2 = new ArrayList();
            Cursor query = contentResolver.query(contentUri, (String[]) null, "media_type=3", (String[]) null, "date_modified ASC");
            if (query == null || !query.moveToNext()) {
                Log.d(TAG, "Cursor null");
                return qVar;
            }
            int columnIndexOrThrow = query.getColumnIndexOrThrow("_id");
            int columnIndexOrThrow2 = query.getColumnIndexOrThrow("bucket_display_name");
            int columnIndexOrThrow3 = query.getColumnIndexOrThrow("bucket_id");
            int columnIndexOrThrow4 = query.getColumnIndexOrThrow("_data");
            int columnIndexOrThrow5 = query.getColumnIndexOrThrow("date_added");
            int columnIndexOrThrow6 = query.getColumnIndexOrThrow("date_modified");
            int columnIndexOrThrow7 = query.getColumnIndexOrThrow("document_id");
            int columnIndexOrThrow8 = query.getColumnIndexOrThrow("mime_type");
            int columnIndexOrThrow9 = query.getColumnIndexOrThrow("_display_name");
            int columnIndexOrThrow10 = query.getColumnIndexOrThrow("owner_package_name");
            int columnIndexOrThrow11 = query.getColumnIndexOrThrow("relative_path");
            int columnIndexOrThrow12 = query.getColumnIndexOrThrow("_size");
            int columnIndexOrThrow13 = query.getColumnIndexOrThrow("relative_path");
            int columnIndexOrThrow14 = query.getColumnIndexOrThrow("media_type");
            ArrayList arrayList3 = arrayList2;
            int columnIndexOrThrow15 = query.getColumnIndexOrThrow("parent");
            int i7 = 50;
            int i8 = columnIndexOrThrow12;
            int i9 = columnIndexOrThrow14;
            while (true) {
                long j5 = query.getLong(columnIndexOrThrow);
                String string = query.isNull(columnIndexOrThrow2) ? null : query.getString(columnIndexOrThrow2);
                String string2 = query.isNull(columnIndexOrThrow3) ? null : query.getString(columnIndexOrThrow3);
                String string3 = query.isNull(columnIndexOrThrow4) ? null : query.getString(columnIndexOrThrow4);
                Long valueOf = query.isNull(columnIndexOrThrow5) ? null : Long.valueOf(query.getLong(columnIndexOrThrow5));
                Long valueOf2 = query.isNull(columnIndexOrThrow6) ? null : Long.valueOf(query.getLong(columnIndexOrThrow6));
                String string4 = query.isNull(columnIndexOrThrow7) ? null : query.getString(columnIndexOrThrow7);
                String string5 = query.isNull(columnIndexOrThrow8) ? null : query.getString(columnIndexOrThrow8);
                String string6 = query.isNull(columnIndexOrThrow9) ? null : query.getString(columnIndexOrThrow9);
                String string7 = query.isNull(columnIndexOrThrow10) ? null : query.getString(columnIndexOrThrow10);
                String string8 = query.isNull(columnIndexOrThrow11) ? null : query.getString(columnIndexOrThrow11);
                int i10 = columnIndexOrThrow;
                int i11 = i8;
                int i12 = i10;
                Integer valueOf3 = query.isNull(i11) ? null : Integer.valueOf(query.getInt(i11));
                String string9 = query.isNull(columnIndexOrThrow13) ? null : query.getString(columnIndexOrThrow13);
                int i13 = i9;
                int i14 = i11;
                int i15 = i13;
                String string10 = query.isNull(i15) ? null : query.getString(i15);
                if (query.isNull(columnIndexOrThrow15)) {
                    i5 = columnIndexOrThrow15;
                    str = null;
                } else {
                    i5 = columnIndexOrThrow15;
                    str = query.getString(columnIndexOrThrow15);
                }
                Uri withAppendedId = ContentUris.withAppendedId(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, j5);
                int i16 = i15;
                K.m(withAppendedId, "withAppendedId(...)");
                arrayList = arrayList3;
                arrayList.add(new MediaStoreFile(j5, string, string2, withAppendedId.getPath(), string3, valueOf, valueOf2, string4, string5, string6, string7, string8, valueOf3, string9, string10, str));
                if (!query.moveToNext()) {
                    break;
                }
                int i17 = i7 - 1;
                if (i7 <= 0) {
                    break;
                }
                arrayList3 = arrayList;
                i7 = i17;
                columnIndexOrThrow15 = i5;
                columnIndexOrThrow = i12;
                i8 = i14;
                i9 = i16;
            }
            query.close();
            return arrayList;
        }
        Uri uri = MediaStore.Images.Media.EXTERNAL_CONTENT_URI;
        return qVar;
    }
}

File: app/src/main/java/com/hawkshaw/library/features/media/files/FileExplorerMediaStoreKt.kt
package com.hawkshaw.library.features.media.files

import android.content.ContentResolver
import android.content.ContentUris
import android.database.Cursor
import android.net.Uri
import android.os.Build
import android.provider.MediaStore
import android.util.Log // Already imported
import java.util.ArrayList

private const val TAG = "FileExplorerMediaStoreKt"

/**
 * Retrieve list of files from MediaStore
 */
fun getMediaStoreFiles(contentResolver: ContentResolver): List<MediaStoreFile> {
    Log.d(TAG, "[DEBUG] getMediaStoreFiles called.")
    // Return empty list for unsupported Android versions
    if (Build.VERSION.SDK_INT < 29) {
        Log.w(TAG, "[DEBUG] Android version ${Build.VERSION.SDK_INT} is less than 29. Returning empty list.")
        return emptyList()
    }
    Log.d(TAG, "[DEBUG] Android version ${Build.VERSION.SDK_INT} is supported.")

    val contentUri = MediaStore.Files.getContentUri("external")
    Log.d(TAG, "[DEBUG] Using content URI: $contentUri")
    val result = ArrayList<MediaStoreFile>()
    
    // Query for document files (media_type=3)
    val selection = "media_type=3"
    val sortOrder = "date_modified ASC"
    Log.d(TAG, "[DEBUG] Querying MediaStore with selection: '$selection', sortOrder: '$sortOrder'")
    val cursor = contentResolver.query(
        contentUri,
        null, // Projection, null means all columns
        selection,
        null, // Selection args
        sortOrder
    )
    
    if (cursor == null) {
        Log.d(TAG, "[DEBUG] Cursor is null. Returning empty list.")
        // Original Log.d(TAG, "Cursor null") - kept for consistency if needed, but [DEBUG] is more specific
        return emptyList()
    }

    if (!cursor.moveToNext()) {
        Log.d(TAG, "[DEBUG] Cursor is empty (moveToNext returned false). Closing cursor and returning empty list.")
        cursor.close()
        return emptyList()
    }
    Log.d(TAG, "[DEBUG] Cursor obtained and is not empty. Proceeding to extract data.")

    // Get column indices
    Log.d(TAG, "[DEBUG] Getting column indices.")
    val idIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns._ID)
    val bucketNameIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.BUCKET_DISPLAY_NAME)
    val bucketIdIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.BUCKET_ID)
    val dataIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.DATA)
    val dateAddedIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.DATE_ADDED)
    val dateModifiedIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.DATE_MODIFIED)
    val documentIdIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.DOCUMENT_ID)
    val mimeTypeIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.MIME_TYPE)
    val displayNameIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.DISPLAY_NAME)
    val ownerPackageNameIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.OWNER_PACKAGE_NAME)
    val relativePathIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.RELATIVE_PATH)
    val sizeIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.SIZE)
    val mediaTypeIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.MEDIA_TYPE)
    val parentIndex = cursor.getColumnIndexOrThrow(MediaStore.Files.FileColumns.PARENT)
    Log.d(TAG, "[DEBUG] Column indices obtained successfully.")

    // Limit the number of results to 50 for performance
    var counter = 0
    val maxItems = 50
    Log.d(TAG, "[DEBUG] Starting to iterate through cursor. Max items: $maxItems")

    do {
        // Extract data from the cursor
        val id = cursor.getLong(idIndex)
        val displayName = if (cursor.isNull(displayNameIndex)) null else cursor.getString(displayNameIndex)
        Log.d(TAG, "[DEBUG] Processing item $counter: ID=$id, DisplayName=$displayName")

        val bucketName = if (cursor.isNull(bucketNameIndex)) null else cursor.getString(bucketNameIndex)
        val bucketId = if (cursor.isNull(bucketIdIndex)) null else cursor.getString(bucketIdIndex)
        val data = if (cursor.isNull(dataIndex)) null else cursor.getString(dataIndex)
        val dateAdded = if (cursor.isNull(dateAddedIndex)) null else cursor.getLong(dateAddedIndex)
        val dateModified = if (cursor.isNull(dateModifiedIndex)) null else cursor.getLong(dateModifiedIndex)
        val documentId = if (cursor.isNull(documentIdIndex)) null else cursor.getString(documentIdIndex)
        val mimeType = if (cursor.isNull(mimeTypeIndex)) null else cursor.getString(mimeTypeIndex)
        // displayName already extracted above
        val ownerPackageName = if (cursor.isNull(ownerPackageNameIndex)) null else cursor.getString(ownerPackageNameIndex)
        val relativePath = if (cursor.isNull(relativePathIndex)) null else cursor.getString(relativePathIndex)
        val size = if (cursor.isNull(sizeIndex)) null else cursor.getInt(sizeIndex)
        val mediaType = if (cursor.isNull(mediaTypeIndex)) null else cursor.getString(mediaTypeIndex)
        val parent = if (cursor.isNull(parentIndex)) null else cursor.getString(parentIndex)

        // Create a content URI for this file
        val contentUriForItem = ContentUris.withAppendedId(contentUri, id)
        Log.d(TAG, "[DEBUG] Item $counter: Content URI for item: $contentUriForItem")

        // Create a MediaStoreFile and add it to the results
        val file = MediaStoreFile(
            id = id,
            bucketName = bucketName,
            bucketId = bucketId,
            contentUri = contentUriForItem.toString(),
            data = data,
            dateAdded = dateAdded,
            dateModified = dateModified,
            documentId = documentId,
            mime = mimeType,
            name = displayName,
            ownerPackageName = ownerPackageName,
            relativePath = relativePath,
            size = size,
            title = displayName, // Using displayName as title
            mediaType = mediaType,
            parent = parent
        )
        Log.d(TAG, "[DEBUG] Item $counter: Created MediaStoreFile object: $file")

        result.add(file)
        counter++
        
    } while (cursor.moveToNext() && counter < maxItems)
    
    Log.d(TAG, "[DEBUG] Finished iterating through cursor. Total items processed: $counter")
    cursor.close()
    Log.d(TAG, "[DEBUG] Cursor closed. Returning ${result.size} files.")
    return result
}
